commcare_connect/form_receiver/processor.py

```diff
@@ -146,7 +146,7 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
     deliver_unit = get_or_create_deliver_unit(app, deliver_unit_block)
     access = OpportunityAccess.objects.get(opportunity=opportunity, user=user)
     counts = (
-        UserVisit.objects.filter(opportunity=opportunity, user=user)
+        UserVisit.objects.filter(opportunity=opportunity, user=user, deliver_unit=deliver_unit)
```

Comment:
I think this logic might work, but it seems not quite right. Ideally we would be checking `CompletedWork` since that is where the limit is, but this seems fine for now.